### PR TITLE
handler for the gitee webhook format.

### PR DESCRIPTION
**Pull request by lly835**
https://gitee.com is the most popular git server in China.
fix https://github.com/spring-cloud/spring-cloud-config/issues/898

### CODE REVIEW

File: spring-cloud-config-monitor/src/test/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractorTests.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.HttpHeaders;
+
+import java.util.Map;
+
+import static org.junit.Assert.*;
```
Comment (ryanjbaxter) [inline spring-cloud-config-monitor/src/test/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractorTests.java]: no `*` imports

File: spring-cloud-config-monitor/src/test/java/org/springframework/cloud/config/monitor/EnvironmentMonitorAutoConfigurationTests.java [hunk 2]
```diff
@@ -16,11 +16,7 @@
 
 package org.springframework.cloud.config.monitor;
 
-import java.util.Collection;
```
Comment (ryanjbaxter) [inline spring-cloud-config-monitor/src/test/java/org/springframework/cloud/config/monitor/EnvironmentMonitorAutoConfigurationTests.java]: make sure your import order is correct

File: spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ */
+@Order(Ordered.LOWEST_PRECEDENCE - 100)
+public class GiteePropertyPathNotificationExtractor
+		implements PropertyPathNotificationExtractor {
+
+	@Override
+	public PropertyPathNotification extract(MultiValueMap<String, String> headers,
+			Map<String, Object> request) {
+		if ("Push Hook".equals(headers.getFirst("x-git-oschina-event"))) {
```
Comment (ryanjbaxter) [inline spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java]: spacing looks off
also i think some of these strings would be good to have as `private static final` vars

File: spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/EnvironmentMonitorAutoConfiguration.java [hunk 4]
```diff
@@ -16,15 +16,15 @@
 
 package org.springframework.cloud.config.monitor;
 
-import java.util.List;
```
Comment (ryanjbaxter) [inline spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/EnvironmentMonitorAutoConfiguration.java]: please use the correct import order

**Comment by lly835** [comment]
@ryanjbaxter review again.

**Comment by ryanjbaxter** [comment]
@lly835 I dont see where you addressed any of my comments

**Comment by lly835** [comment]
@ryanjbaxter https://github.com/spring-cloud/spring-cloud-config/pull/901/files

**Comment by spencergibb** [comment]
There has been no update to this pull request

**Comment by lly835** [comment]
@spencergibb @ryanjbaxter  i don't know "make sure your import order is correct".I already answered review. I'm in China. What's the quickest way to communicate?

**Comment by ryanjbaxter** [comment]
@lly835 see the eclipse formatting files listed in the contributor guidelines  (this works with Eclipse or Intellij).  Once you have the formatting files configured in your IDE the import order will be properly set.  https://github.com/spring-cloud/spring-cloud-config/blob/master/.github/CONTRIBUTING.md#code-conventions-and-housekeeping

**Comment by lly835** [comment]
@ryanjbaxter  import order is correct now.

File: spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java (hunk 3 above)
Comment (ryanjbaxter) [reply spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java]: Can you fix ^^^

**Comment by spencergibb** [comment]
@lly835 docs are here https://github.com/spring-cloud/spring-cloud-config/blob/master/docs/src/main/asciidoc/spring-cloud-config.adoc

File: spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+	private static final String HEADERS_KEY = "x-git-oschina-event";
+
+	private static final String HEADERS_VALUE = "Push Hook";
+
+	@Override
+	public PropertyPathNotification extract(MultiValueMap<String, String> headers,
+			Map<String, Object> request) {
+		if (HEADERS_VALUE.equals(headers.getFirst(HEADERS_KEY))) {
+            if (request.get("commits") instanceof Collection &&
```
Comment (spencergibb) [inline spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java]: We use tabs not spaces

Comment (lly835) [reply spring-cloud-config-monitor/src/main/java/org/springframework/cloud/config/monitor/GiteePropertyPathNotificationExtractor.java]: has been modified

**Comment by ryanjbaxter** [comment]
@lly835 I dont see any documentation changes

**Comment by lly835** [comment]
@spencergibb docs/src/main/asciidoc/spring-cloud-config.adoc updated

**Pull request merged**